site-src/geps/gep-709.md

```diff
@@ -0,0 +1,260 @@
+// Gateway API.
+type ReferencePolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of ReferencePolicy.
+    Spec ReferencePolicySpec `json:"spec,omitempty"`
+}
+
+
+// ReferencePolicySpec identifies a cross namespace relationship that is trusted
+// for Gateway API.
+type ReferencePolicySpec struct {
+    // From describes the trusted namespaces and kinds that can reference the
+    // resources described in "To".
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinItems=1
+    From []ReferencePolicyFrom `json:"from"`
+
+    // To describes the resources that may be referenced by the resources
+    // described in "From".
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinItems=1
+    To []ReferencePolicyTo `json:"to"`
+}
+
+// ReferencePolicyFrom describes trusted namespaces and kinds.
+type ReferencePolicyFrom struct {
+    // Group is the group of the referrent.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    Group string `json:"group"`
+
+    // Kind is kind of the referrent.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    Kind string `json:"kind"`
+
+    // Namespace is the namespace of the referrent.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    Namespace string `json:"namespace,omitempty"`
+}
+
+// ReferencePolicyTo describes trusted kinds.
+type ReferencePolicyTo struct {
+    // Group is the group of the referrent.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    Group string `json:"group"`
+
+    // Kind is kind of the referrent.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    Kind string `json:"kind"`
+}
+```
+
+
+### Benefits
+
+* Conceptually similar to NetworkPolicy.
+* A separate resource enables admins to restrict who can allow cross namespace
+  references.
+* Provides consistent way to control references to any resource from a Route.
+* Can be extended in the future for additional use cases.
+* A single ReferencePolicy resource can be used for a namespace in place of
+  separate handshake config on each Service or Route resource.
+
+#### Exceptions
+If traffic is originating from the same location as the configured Route (such
+as sidecars in some mesh implementations) implementations may choose to ignore
+ReferencePolicy. This should only be done if:
+* Other mechanisms like NetworkPolicy can be used to effectively limit
+  cross-namespace references.
+* The implementation clearly documents that ReferencePolicy is not honored.
```

Comment:
Small change to make this clearer:
```suggestion
ReferencePolicy. This may only be done if:
* Other mechanisms like NetworkPolicy are used to effectively limit
  cross-namespace references.
* The implementation must clearly document that ReferencePolicy is not honored.
```

I think that "may" is okay here because we are following it up with clarifying "must" clauses straight afterwards.

Reply:
MAY, MUST, SHOULD should be capitalized so it's obvious we are using them in LOUD RFC TALK.

Reply:
Yes, that's a great way to do it.

Reply:
Now that the CVE is live, I've reworked this entire section, including "LOUD RFC TALK" :), PTAL.